src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/KestrelHttpParser.cs

```
@@ -162,6 +185,10 @@ public KestrelHttpParser(IKestrelTrace log)
                                     RejectRequestLine(span);
                                 }
                             }
+                            else if (ch == 0)
+                            {
+                                RejectRequestLine(span);
```

Fail early when possible. \0 is never acceptable on the request line.